src/openai/lib/azure.py

```diff
@@ -0,0 +1,210 @@
+from __future__ import annotations
+
+import os
+from typing import Mapping, overload
+
+import httpx
+
+from .._types import NOT_GIVEN, Timeout, NotGiven
+from .._utils import is_mapping
+from .._client import OpenAI, AsyncOpenAI
+from .._models import FinalRequestOptions
+from .._exceptions import OpenAIError
+from .._base_client import DEFAULT_MAX_RETRIES, BaseClient
+
+_deployments_endpoints = set(["/completions", "/chat/completions"])
+
+
+class BaseAzureClient(BaseClient):
+    def _build_request(
+        self,
+        options: FinalRequestOptions,
+    ) -> httpx.Request:
+        if options.url in _deployments_endpoints and is_mapping(options.json_data):
+            model = options.json_data.get("model")
+            if model is not None and not "/deployments" in str(self.base_url):
+                options.url = f"/deployments/{model}{options.url}"
+
+        return super()._build_request(options)
+
+
+class AzureOpenAI(BaseAzureClient, OpenAI):
+    @overload
+    def __init__(
+        self,
+        *,
+        api_version: str,
+        endpoint: str,
```

Comment:
In isolation, think `endpoint` is a good parameter name (it matches the terminology in the Azure portal as well as following our client library naming convention - although the former is more important than the latter here).